packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -383,7 +389,21 @@ function hydrateChildren(
         }
     }
 
-    if (nextNode) {
+    const useCommentsForBookends = isAPIFeatureEnabled(
+        APIFeature.USE_COMMENTS_FOR_FRAGMENT_BOOKENDS,
+        owner.apiVersion
+    );
+    if (
+        // If 1) comments are used for bookends, and 2) we're not expecting additional siblings,
+        // and 3) there exists an additional sibling, that's a hydration failure.
```

Comment:
Reading this comment makes me think the condition below should be
```javascript
useCommentsForBookends && !expectAddlSiblings && nextNode
```
and not what it currently says on line 404:
```javascript
(!useCommentsForBookends || !expectAddlSiblings) && nextNode
```
Am I missing something or does the comment or conditional statement need updating?

Reply:
These are the scenarios:

- `useCommentsForBookends` is `false`. In this case, we don't care if `expectAddlSiblings` is `true` or `false`. All we care about is whether there is a `nextNode` (preserving pre-existing behavior).
- `useCommentsForBookends` is `true`. In this case, we _do_ care about the value of `expectAddlSiblings`.
    - `expectAddlSiblings` is `true`. In this case, we don't care if there's a next node, because a next node is permitted. The parenthesized expression would be `false` and the larger condition would be `false && nextNode`, meaning we wouldn't expect the if-block's content
    - `expectAddlSiblings` is `false`. In this case, we _do_ care about whether there's a next node. Because if there's a next node and we're not expecting it, that's a hydration failure. The parenthesized expression evaluates to `true` and the larger condition would be `true && nextNode`.

I think this all lines up with what's in the comment. It's just a weird looking `if` condition. But if that still doesn't seem right, I can try to rephrase the comment to be clearer.